src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const historyClones = [];
+  const currentSt = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'clear':
+        for (const i in currentSt) {
+          delete currentSt[i];
+        }
+        break;
+
+      case 'addProperties':
+        Object.assign(currentSt, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const i of keysToRemove) {
+          if (currentSt[i]) {
+            delete currentSt[i];
```

Comment:
The check if 'currentSt[i]' exists before deleting is unnecessary. The delete operator will not throw an error if the property does not exist.